File: backend/schemas/condition_group.py
"""
组合条件 Schema 定义
"""
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime


class ConditionGroupCreate(BaseModel):
    """创建组合条件组"""
    group_name: str
    logic_type: str = "AND"  # AND 或 OR
    parent_id: Optional[int] = None
    priority: str = "warning"
    description: Optional[str] = None
    condition_ids: List[int] = []  # 创建时直接关联条件


class ConditionGroupUpdate(BaseModel):
    """更新组合条件组"""
    group_name: Optional[str] = None
    logic_type: Optional[str] = None
    priority: Optional[str] = None
    is_enabled: Optional[bool] = None
    description: Optional[str] = None


class ConditionItemCreate(BaseModel):
    """添加条件项"""
    condition_id: int
    sort_order: int = 0


class ConditionGroupTreeNode(BaseModel):
    """组合条件树节点"""
    id: int
    group_key: str
    group_name: str
    logic_type: str
    priority: str
    is_enabled: bool
    description: Optional[str] = None
    conditions: List[dict] = []  # 包含的条件列表
    subgroups: List['ConditionGroupTreeNode'] = []  # 子分组

    class Config:
        from_attributes = True


class ConditionGroupList(BaseModel):
    """组合条件列表项"""
    id: int
    group_key: str
    group_name: str
    logic_type: str
    priority: str
    is_enabled: bool
    parent_id: Optional[int] = None
    description: Optional[str] = None
    condition_count: int = 0
    subgroup_count: int = 0
    created_at: datetime
    updated_at: datetime

File: frontend/src/api/conditionGroup.ts
/**
 * 组合条件 API
 */
import request from '@/utils/request'

// ==================== 类型定义 ====================

export interface ConditionGroup {
  id: number
  group_key: string
  group_name: string
  logic_type: 'AND' | 'OR'
  priority: string
  is_enabled: boolean
  parent_id: number | null
  description: string | null
  condition_count: number
  subgroup_count: number
  created_at: string
  updated_at: string
}

export interface ConditionGroupTreeNode {
  id: number
  group_key: string
  group_name: string
  logic_type: 'AND' | 'OR'
  priority: string
  is_enabled: boolean
  description: string | null
  conditions: ConditionItem[]
  subgroups: ConditionGroupTreeNode[]
}

export interface ConditionItem {
  item_id?: number
  condition_id: number
  condition_key: string
  condition_name: string
  priority: string
  sort_order?: number
}

export interface CreateGroupParams {
  group_name: string
  logic_type: 'AND' | 'OR'
  parent_id?: number
  priority?: string
  description?: string
  condition_ids?: number[]
}

export interface UpdateGroupParams {
  group_name?: string
  logic_type?: 'AND' | 'OR'
  priority?: string
  is_enabled?: boolean
  description?: string
}

// ==================== API 函数 ====================

/** 获取组合条件列表 */
export async function fetchConditionGroups(parentId?: number): Promise<ConditionGroup[]> {
  const params = parentId !== undefined ? { parent_id: parentId } : {}
  return request.get('/warning/groups', { params })
}

/** 获取组合条件树 */
export async function fetchConditionGroupTree(): Promise<ConditionGroupTreeNode[]> {
  return request.get('/warning/groups/tree')
}

/** 获取单个组合条件详情 */
export async function fetchConditionGroup(id: number): Promise<ConditionGroupTreeNode> {
  return request.get(`/warning/groups/${id}`)
}

/** 创建组合条件 */
export async function createConditionGroup(data: CreateGroupParams): Promise<{ id: number; group_key: string }> {
  return request.post('/warning/groups', data)
}

/** 更新组合条件 */
export async function updateConditionGroup(id: number, data: UpdateGroupParams): Promise<void> {
  return request.put(`/warning/groups/${id}`, data)
}

/** 删除组合条件 */
export async function deleteConditionGroup(id: number): Promise<void> {
  return request.delete(`/warning/groups/${id}`)
}

/** 添加条件项 */
export async function addConditionItem(groupId: number, conditionId: number, sortOrder?: number): Promise<{ id: number }> {
  return request.post(`/warning/groups/${groupId}/items`, { condition_id: conditionId, sort_order: sortOrder })
}

/** 移除条件项 */
export async function removeConditionItem(groupId: number, itemId: number): Promise<void> {
  return request.delete(`/warning/groups/${groupId}/items/${itemId}`)
}

/** 创建子分组 */
export async function createSubgroup(parentId: number, data: CreateGroupParams): Promise<{ id: number; group_key: string }> {
  return request.post(`/warning/groups/${parentId}/subgroups`, data)
}